libzfs-sys/build.rs

```diff
@@ -91,15 +90,55 @@ fn main() {
         .whitelist_function("zfs_expand_proplist")
         .whitelist_function("zfs_prop_to_name")
         .whitelist_function("zfs_validate_name")
-        .whitelist_function("zprop_free_list")
-        .clang_arg("-I/usr/lib/gcc/x86_64-redhat-linux/4.8.2/include/")
-        .clang_arg("-I/usr/src/zfs-0.7.9/lib/libspl/include/")
-        .clang_arg("-I/usr/src/zfs-0.7.9/include/")
-        .generate()
-        .expect("Unable to generate bindings");
+        .whitelist_function("zprop_free_list");
+
+    let bindings = if cfg!(target_os = "freebsd") {
+        println!("cargo:rustc-link-lib=zfs");
+        // a subset of include paths in cddl/sbin/zfs/Makefile
+        bindings
+            .clang_arg("-I/usr/src/cddl/compat/opensolaris/include")
+            .clang_arg("-I/usr/src/cddl/compat/opensolaris/lib/libumem")
+            .clang_arg("-I/usr/src/cddl/contrib/opensolaris/head")
+            .clang_arg("-I/usr/src/cddl/contrib/opensolaris/lib/libuutil/common")
+            .clang_arg("-I/usr/src/cddl/contrib/opensolaris/lib/libzfs/common")
+            .clang_arg("-I/usr/src/cddl/contrib/opensolaris/lib/libzpool/common")
+            .clang_arg("-I/usr/src/sys/cddl/compat/opensolaris")
+            .clang_arg("-I/usr/src/sys/cddl/contrib/opensolaris/uts/common")
+            .clang_arg("-I/usr/src/sys/cddl/contrib/opensolaris/uts/common/fs/zfs")
+            .clang_arg("-I/usr/src/sys/cddl/contrib/opensolaris/common/zfs")
+    } else {
+        let libzfs = pkg_config::Config::new().probe("libzfs").unwrap();
+        list_gcc_include_paths()
+            .chain(libzfs.include_paths)
+            .fold(bindings, |bindings, include_path| {
+                let flag = format!("-I{}", include_path.to_string_lossy());
+                bindings.clang_arg(flag)
+            })
+        // include directory for zfsonlinux/zfs master branch
+            .clang_arg("-I/usr/src/zfs-0.7.0/include")
```

Comment:
This will get a bit farther if it is changed to:

`.clang_arg("-I/usr/src/zfs-0.7.0/include")`